docs/development_suite/api-console/api-design/microservice-containers.md

```diff
@@ -0,0 +1,88 @@
+---
+id: microservice-containers
+title: Microservice Containers
+sidebar_label: Microservice Containers
+---
+
+# Microservice Containers
+
+A microservice in Console is a collection of one or more containers that solve a single yet well defined business scope. 
+
+In general microservices define a single docker image and thus a single **main container**.
+
+Sometimes tough, it may be useful to run another container alongside the main container to add some capabilities. Usually these kind of containers are used to avoid cluttering the main service with helper code. These containers are known as **sidecar containers**.
+
+## Main Container
+
+When creating a new microservice in Console, you can define a single docker image that will be used for the **main container** of your microservice. This container should contain the core business feature of the microservice itself. 
+
+## Sidecar Containers
+
+For several reasons although, you may want to extend the main container functionalities **without modifying** the business logic of your application.
+
+To better understand this need, let's imagine this common scenario.  
+You have a core microservice that processes user orders. This microservice is responsible for validating the order, calculating the total amount, and then sending it to the fulfillment center.
+
+However, you now have a new requirement: adding a logging functionality to your microservice to monitor every order that is being processed.  
+You could simply add the logging code to the core service source code, but sometimes this option is not viable, to name a few reasons:
+- the service may be maintained by another team
+- modifying the legacy core service may introduce errors
+
+For these reasons the sidecar-pattern comes very handy. You can develop a lightweight service that specializes in logging requests to the main application and store these logs in a centralized logging system.  
+You can then deploy this logging service along with the main application on the same host. This ensures that the sidecar and main service containers are always running on the same machine.
+
+In this section we will go deeper in how you can manage this kind of containers for your microservices.
+
+
+## Microservice detail page
+
+The containers table in the microservice detail page gives an immediate overview of all the containers that your microservice is made of, indicating which is the **main container** and which is the **sidecar** one.
+
+The table shows a summary of the containers resource request and limits for CPU and memory as well as which configmaps and secrets are used by which container. From here is also possible to add and delete sidecars to the current microservice.
+
+![Microservice Containers](img/microservice-containers/containers-table.png)
+
+### Exposing a specific container
```

Comment:
maybe we could add a brief reference to the cases in which exposure management is limited (e.g. special sidecars, no container ports...)